contracts/WildcardSteward_v2.sol

```diff
@@ -450,13 +470,57 @@ contract WildcardSteward_v2 is Initializable {
         emit RemainingDepositUpdate(patron, deposit[patron]);
     }
 
-    function buy(uint256 tokenId, uint256 _newPrice, uint256 _deposit)
+    function changeAuctionParamters(
+        uint256 _auctionStartPrice,
+        uint256 _auctionEndPrice,
+        uint256 _auctionLength
+    ) external onlyAdmin {
+        require(
+            _auctionStartPrice > _auctionEndPrice,
+            "Auction value must decrease over time"
+        );
+        require(_auctionLength >= 86400, "Auction should last at least day");
+
+        auctionStartPrice = _auctionStartPrice;
+        auctionEndPrice = _auctionEndPrice;
+        auctionLength = _auctionLength;
+    }
+
+    // NOTE: Need to distinguish been previously owned and foreclosed vs first sale items.
+    //
+    function _auctionPrice(uint256 tokenId) internal returns (uint256) {
+        // HERE we need to be careful. Check whether timeLastCollected[tokenId] is actually now due to collect Patronage modifer being called
+        uint256 auctionEnd = timeLastCollected[tokenId].add(auctionLength);
+        if (now >= auctionEnd) {
+            return auctionEndPrice;
+        } else {
+            // (y2-y1)/(x2-x1)
+            uint256 slope = (auctionEndPrice.sub(auctionStartPrice)).div(
```

Comment:
we will need to invert this value if we are using uint. For example `auctionEndPrice.sub(auctionStartPrice)` will be negative. This is fine, we can swap it to be `(y1-y2)/(x2-x1) ==  -(y2-y1)/(x2-x1)` and then in the "intercept" equation we use add (`uint256 intercept = auctionEnd.add(slope.mul(auctionEndPrice)); `), basically `c = y + (-m)x`. Also the last step you would need to do `intercept.sub(slope.mul(now))`. And because of the check at the begining that now is less than the auction end, this will never be negative.

What I'm thinking, Not sure it is better or anything. But what if we fix the slope (or the negative slope) to be decrease price by 'value of slope' wei every second.
Then calculating the current price before it goes below the min price is easy.
So in this case, the important computation is  we need to calculate the auctionEnd; where in the equation `y=mx+c` `c` is startingPrice, `m` is the slope (or the negative of the price reduction in wei per second), y is the min price; then x pops out as the `auctionEnd`. I didn't check, but my gut feeling is that this would be more gas efficient because I think there are less maths operations happening.